pkg/ccl/crosscluster/producer/event_stream.go

```diff
@@ -160,6 +160,9 @@ func (s *eventStream) Start(ctx context.Context, txn *kv.Txn) (retErr error) {
 		// stream.
 		opts = append(opts, rangefeed.WithOriginIDsMatching(0))
 	}
+	if s.spec.WithFiltering {
```

Comment:
nit: this couple be simplified, without a branch: `rangefeed.WithFiltering(s.SpecWithFiltering)`